deepchecks/utils/validation.py

```diff
@@ -23,3 +24,48 @@ def model_type_validation(model: t.Any):
             'Model must inherit from one of supported '
             'models: sklearn.base.BaseEstimator or CatBoost'
         )
+
+
+def validate_model(dataset: 'base.Dataset', model: t.Any):
+    """Check model is able to predict on the dataset.
+
+    Raise:
+        DeepchecksValueError: if dataset does not match model
+    """
+    model_type_validation(model)
+
+    error_message = (
+        'In order to evaluate model correctness we need not empty dataset '
+        'with the same set of features that was used to fit the model! {0}'
+    )
+
+    features = dataset.features_columns()
+    features_names = set(dataset.features())
+    model_features = getattr(model, 'feature_names_in_', None)
+
+    if features is None:
+        raise errors.DeepchecksValueError(error_message.format(
+            'But function received dataset without feature columns!'
+        ))
+
+    if len(features) == 0:
+        raise errors.DeepchecksValueError(error_message.format(
+            'But function received empty dataset!'
+        ))
+
+    try:
+        model_features = set(model_features) # type: ignore
+        if model_features != features_names:
+            raise errors.DeepchecksValueError(error_message.format(
+                'But function received dataset with a different set of features!'
+            ))
+    except (TypeError, ValueError):
+        # in case if 'model.feature_names_in_' was None or not iterable
+        pass
+
+    try:
+        model.predict(features.head(1))
+    except Exception as exc:
```

Comment:
Open question - should we be catching ValueError?

Reply:
I saw that sklearn always raises ValueError, but I think there is nothing bad in catching more general error type/ just in case if in some rare situation it will raise something different

Reply:
I can go with that